vllm/worker/worker.py

```diff
@@ -274,6 +279,17 @@ def initialize_cache(self, num_gpu_blocks: int,
         self._init_cache_engine()
         self._warm_up_model()
 
+        # Initialize the buffer for swapping in/out blocks.
+        max_num_blocks = max(num_gpu_blocks, num_cpu_blocks)
+        self.blocks_to_swap_in_buffer = torch.zeros((max_num_blocks, 2),
+                                                    dtype=torch.int64,
+                                                    device="cpu",
+                                                    pin_memory=True)
+        self.blocks_to_swap_out_buffer = torch.zeros((max_num_blocks, 2),
+                                                     dtype=torch.int64,
+                                                     device="cpu",
+                                                     pin_memory=True)
```

Comment:
some systems do not have pin memory (notably, WSL). we need to take care of that. otherwise this PR LGTM.

Reply:
WSL does not support UVA, either. You can use `is_pin_memory_available` to determine if this optimization can be used.